frontend/packages/dev-console/src/components/topology/Topology.tsx

```diff
@@ -9,24 +11,65 @@ import {
   BaseEdge,
   VisualizationProvider,
   Model,
+  GraphModel,
+  NodeModel,
   BOTTOM_LAYER,
   GROUPS_LAYER,
   DEFAULT_LAYER,
   TOP_LAYER,
   SelectionEventListener,
   SELECTION_EVENT,
+  NODE_POSITIONED_EVENT,
+  GRAPH_POSITION_CHANGE_EVENT,
 } from '@patternfly/react-topology';
-import { useQueryParams } from '@console/shared';
+import { STORAGE_PREFIX, useQueryParams } from '@console/shared';
 import { useExtensions } from '@console/plugin-sdk';
+import { RootState } from '@console/internal/redux';
 import { isTopologyComponentFactory, TopologyComponentFactory } from '../../extensions/topology';
 import { SHOW_GROUPING_HINT_EVENT, ShowGroupingHintEventListener } from './topology-types';
+import { getTopologyGraphModel, setTopologyGraphModel } from './redux/action';
 import { COLA_LAYOUT, layoutFactory } from './layouts/layoutFactory';
 import { componentFactory } from './components';
 import { odcElementFactory } from './elements';
 import TopologyControlBar from './TopologyControlBar';
 
 import './Topology.scss';
 
+const TOPOLOGY_LAYOUT_LOCAL_STORAGE_KEY = `${STORAGE_PREFIX}/topology-layout`;
+const STORED_NODE_LAYOUT_FIELDS = ['id', 'x', 'y', 'collapsed', 'visible', 'style', 'shape'];
+
+type StoredLayout = {
+  nodes: NodeModel[];
+  layout: string;
+};
+
+const getStoredLayouts = (): { [namespace: string]: StoredLayout } => {
+  const storedLayoutsString = localStorage.getItem(TOPOLOGY_LAYOUT_LOCAL_STORAGE_KEY);
+  try {
+    return JSON.parse(storedLayoutsString);
```

Comment:
@jeff-phillips-18 `localStorage.getItem()` returns null for a key which is not existing yet. And JSON.parse(null) returns null. So I think you want to add a fallback here, or? ` || {}`